packages/react-dom/src/__tests__/ReactDOMFiber-test.js

```diff
@@ -27,157 +42,86 @@ describe('ReactDOMFiber', () => {
 
-    ReactDOM.render(<Box value={10} />, container);
+    await act(async () => {
+      root.render(<Box value={10} />);
+    });
 
     expect(container.textContent).toEqual('10');
   });
 
-  it('should be called a callback argument', () => {
+  it('should call an effect after mount/update (replacing render callback pattern)', async () => {
+    function Component() {
+      React.useEffect(() => {
+        Scheduler.log('Callback');
+      });
+      return <div>Foo</div>;
+    }
+
     // mounting phase
-    let called = false;
-    ReactDOM.render(<div>Foo</div>, container, () => (called = true));
-    expect(called).toEqual(true);
+    await act(async () => {
+      root.render(<Component />);
+    });
+    assertLog(['Callback']);
 
     // updating phase
-    called = false;
-    ReactDOM.render(<div>Foo</div>, container, () => (called = true));
-    expect(called).toEqual(true);
+    await act(async () => {
+      root.render(<Component />);
+    });
+    assertLog(['Callback']);
   });
 
-  it('should call a callback argument when the same element is re-rendered', () => {
-    class Foo extends React.Component {
-      render() {
-        return <div>Foo</div>;
-      }
+  it('should call an effect when the same element is re-rendered (replacing render callback pattern)', async () => {
+    function Component({prop}) {
+      React.useEffect(() => {
+        Scheduler.log('Callback');
+      });
+      return <div>{prop}</div>;
     }
-    const element = <Foo />;
 
     // mounting phase
-    let called = false;
-    ReactDOM.render(element, container, () => (called = true));
-    expect(called).toEqual(true);
+    await act(async () => {
+      root.render(<Component prop="Foo" />);
+    });
+    assertLog(['Callback']);
 
     // updating phase
-    called = false;
-    ReactDOM.unstable_batchedUpdates(() => {
-      ReactDOM.render(element, container, () => (called = true));
+    await act(async () => {
+      root.render(<Component prop="Bar" />);
     });
-    expect(called).toEqual(true);
+    assertLog(['Callback']);
   });
 
-  it('should render a component returning strings directly from render', () => {
+  it('should render a component returning strings directly from render', async () => {
     const Text = ({value}) => value;
 
-    ReactDOM.render(<Text value="foo" />, container);
+    await act(async () => {
+      root.render(<Text value="foo" />);
+    });
+
     expect(container.textContent).toEqual('foo');
   });
 
-  it('should render a component returning numbers directly from render', () => {
+  it('should render a component returning numbers directly from render', async () => {
     const Text = ({value}) => value;
-
-    ReactDOM.render(<Text value={10} />, container);
+    await act(async () => {
+      root.render(<Text value={10} />);
+    });
 
     expect(container.textContent).toEqual('10');
   });
 
-  it('finds the DOM Text node of a string child', () => {
```

Comment:
I deleted all the `findDOMNode` tests from this file, they still exist in `ReactDOMLegacyFiber`.